### PR TITLE
New package: kubectx-0.9.4

### PR DESCRIPTION
**Pull request by fanyx**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**|**NO**

#### Local build testing
- I built this PR locally for my native architecture, x86_64

New PR for previously abandoned package.
https://github.com/void-linux/void-packages/pull/17088

I'll maintain updates if necessary.

### CODE REVIEW

File: srcpkgs/kubectx/template [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+do_install() {
+	vbin kubectx
+	vbin kubens
+	vinstall completion/kubectx.fish 0644 usr/share/fish/completions
+	vinstall completion/kubens.fish 0644 usr/share/fish/completions
+	vinstall completion/kubectx.zsh 0644 usr/share/zsh/site-functions
+	vinstall completion/kubens.zsh 0644 usr/share/zsh/site-functions
+	vinstall completion/kubectx.bash 0644 usr/share/bash-completion/completions kubectx
+	vinstall completion/kubens.bash 0644 usr/share/bash-completion/completions kubens
```
Comment (classabbyamp) [inline srcpkgs/kubectx/template:22]:
```suggestion
	for bin in kubectx kubens; do
		vbin $bin
		for sh in bash zsh fish; do
			vcompletion completion/$bin.$sh $sh $bin
		done
	done
```

File: srcpkgs/kubectx/template [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Template file for 'kubectx'
+pkgname=kubectx
+version=0.9.4
+revision=1
+build_style=meta
```
Comment (classabbyamp) [inline srcpkgs/kubectx/template:5]: this is not a meta package

**Comment by classabbyamp** [comment]
imo, for this to meet quality requirements, it would need to be the go version, not the bash version:

> Stable versions of `kubectx` and `kubens` are small bash scripts that you can find in this repository.
>
> Starting with v0.9.0, `kubectx` and `kubens` **are now rewritten in Go**. They should work the same way (and we'll keep the bash-based implementations around) but the new features will be added to the new Go programs.

**Review by classabbyamp — CHANGES_REQUESTED** [review]
imo, for this to meet quality requirements, it would need to be the go version, not the bash version:

> Stable versions of `kubectx` and `kubens` are small bash scripts that you can find in this repository.
>
> Starting with v0.9.0, `kubectx` and `kubens` **are now rewritten in Go**. They should work the same way (and we'll keep the bash-based implementations around) but the new features will be added to the new Go programs.

**Comment by fanyx** [comment]
@classabbyamp 
I've used ${worksrc} as i wasn't sure what $PWD is during do_build and do_install.
Is that fine or do you have a suggestion on this?

**Comment by fanyx** [comment]
interesting
builds work just fine on my x86_64 system...

**Comment by fanyx** [comment]
thats what i thought, but it doesnt build when i leave it out...

**Comment by classabbyamp** [comment]
I would also recommend using the go build_style for this, it might make a few things easier

File: srcpkgs/kubectx/template [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+license="Apache-2.0"
+homepage="https://github.com/ahmetb/kubectx"
+distfiles="https://github.com/ahmetb/kubectx/archive/v${version}.tar.gz"
+checksum=91e6b2e0501bc581f006322d621adad928ea3bd3d8df6612334804b93efd258c
+nopie=y
+
+do_build() {
+	go build -o ${worksrc}/bin/kubectx ./cmd/kubectx/
+	go build -o ${worksrc}/bin/kubens ./cmd/kubens/
+}
```
Comment (paper42) [inline srcpkgs/kubectx/template:18]: use the go build style instead

Comment (fanyx) [reply srcpkgs/kubectx/template:18]: i was gonna but it won't work as the repo root doesn't hold a main.go
code is in subdirectories, but those dont hold go.mod files either

Comment (paper42) [reply srcpkgs/kubectx/template]: Look at other templates, this is usually done by setting go_package and go_import_path

Comment (fanyx) [reply srcpkgs/kubectx/template]: ah, go_package was the one i was missing
let me add that

**Comment by fanyx** [comment]
thanks for the help @paper42 
should work as intended now

any suggestions on patching the correct version in, though?
should i keep the patch file and update it with every version manually or can we read the $version from the template during patching somehow?

**Comment by paper42** [comment]
> any suggestions on patching the correct version in, though?
> should i keep the patch file and update it with every version manually or can we read the $version from the template during patching somehow?

`go_ldflags="-X ${go_import_path}/cmd/kubens.version=${version}"` and similar for the second binary

**Comment by fanyx** [comment]
> `go_ldflags="-X ${go_import_path}/cmd/kubens.version=${version}"` and similar for the second binary

unfortunately does not achieve the desired result..

**Comment by fanyx** [comment]
`go_ldflags="-X main.version=v${version}"` did the job just fine.
`go_ldflags="-X ${go_import_path}/cmd/kubectx/version.version=v${version}` unfortunately does not either, no idea why...

**Comment by fanyx** [comment]
i'd suspend this package for `kubectx-bash` #40005 for the time being due to this go version lacking features available in the bash version

**Pull request closed without merging**